lib/parsers/fdx_parser.rb

```diff
@@ -0,0 +1,33 @@
+class FormatParser::FDXParser
+  include FormatParser::IOUtils
+
+  def information_from_io(io)
+    if xml_check(io)
+      file_and_document_type = safe_read(io, 100)
+      file_type, document_type = check_for_document_type(file_and_document_type)
+    else
+      return # Bail if it's not even XML
+    end
+
```

Comment:
Youa re having a bit of a gotofail here - if `check_for_document_type` returns a nil, you are going to return a `:document` FileInformation with a `nil` file_type, in practice you will be capturing _any_ XML document. Including, say, MusicXML and SVG.